api/room/schema_query.py

```diff
@@ -308,6 +325,85 @@ def resolve_all_remote_rooms(self, info, return_all=None):
                         if not re.search(match, room.name)]
         return AllRemoteRooms(rooms=actual_rooms)
 
+    all_available_rooms = graphene.Field(
+        AllAvailableRooms,
+        start_date=graphene.String(required=True),
+        start_time=graphene.String(required=True),
+        end_date=graphene.String(required=True),
+        end_time=graphene.String(required=True),
+        time_zone=graphene.String(required=True),
+        description="Returns available rooms for a specific time \
+            \n- start_time: Start time and date when you want to book room from\
+            [required]\n- end_time: time and date you want to book room upto\
+                [required]\n time_zone: The time zone of the location\
+                    [required]\n location: The location of the office's \
+                        room you want to book"
+    )
+
+    @Auth.user_roles('Admin', 'Super Admin')
+    def resolve_all_available_rooms(self, info, **kwargs):
+
+        time_zone = kwargs['time_zone']
+        empty_string_checker(time_zone)
+        validate_date_input(kwargs['start_date'], kwargs['end_date'])
+        validate_date_input(kwargs['start_time'], kwargs['end_time'])
+        format_range_dates(kwargs['start_date'], kwargs['end_date'])
+        format_range_time(kwargs['start_time'], kwargs['end_time'])
+
+        new_start_time = convert_date(kwargs['start_date'],
+                                      kwargs['start_time'],
+                                      time_zone
+                                      )
+        new_end_time = convert_date(kwargs['end_date'],
```

Comment:
`new_end_time` is not a descriptive variable name. Please describe better what the result of the converted date is.